src/amber/cli/helpers/helpers.cr

```diff
@@ -1,11 +1,19 @@
 module Amber::CLI::Helpers
   def add_routes(pipeline, route)
     routes = File.read("./config/routes.cr")
-    replacement = <<-ROUTES
-    routes :#{pipeline.to_s} do
-    #{route}
-    ROUTES
-    File.write("./config/routes.cr", routes.gsub("routes :#{pipeline.to_s} do", replacement))
+    if pipeline.to_s == "web"
```

Comment:
not liking the hardcoded pipelines `web` and `api` with special logic here.

Reply:
Would you prefer something like this?

```
def add_routes(pipeline, route)
    scope = "routes: #{pipeline.to_s == "web" ? pipeline.to_s ? pipeline.to_s + "\"/api\""} do"
    routes = File.read(".config/routes.cr")
    replacement = <<-ROUTES
    #{scope}
    #{route}
    ROUTES
    File.write(".config/routes.cr", routes.gsub(scope, replacement)
end
```

or do we not want `routes :api, "/api" do` at all?

Reply:
hhmm, good question. I am ok with the `routes :api do` since we have a different set of middleware.  I guess i'm in agreement with @alex-lairan that having the hard coded `/api` prefix is where I'm uncomfortable.  That seems inflexible to me.

Is it necessary to have the "/api" namespace here?  Is that to avoid conflicts?

Reply:
So the problem I was running into with the [amber-vue](https://github.com/seanwatters/amber-vue) and [reactive-amber](https://github.com/seanwatters/reactive-amber) recipes, is that if I create an API with the generator, I have to go manually rewrite the routes file and manually name space the APIs every time I generate a new one. 

If I generate a default API for posts, and want to use a "/posts" route for vue or react router, I can't because it's been taken.

Reply:
The primary goal is to make it easier for people to use progressive frontend frameworks, with Amber, without running a separate node project or using separate domain for their API (or manually setting the "/api" scope every time you create a new JSON API).

Reply:
>  I have to go manually rewrite the routes file and manually name space the APIs every time I generate a new one.

That is a bug.  You shouldn't have to do that. You should only have to set the scope once.  When the resource is added to the `api` pipeline, it shouldn't overwrite the scope.  If that is what is happening, we should fix that instead of adding a hard coded scope.

Reply:
@seanwatters I created a PR to fix the bug.  https://github.com/amberframework/amber/pull/1165

Let me know if this resolves your issue.

Reply:
> @seanwatters I created a PR to fix the bug. #1165
> 
> Let me know if this resolves your issue.

That looks great! I think that actually solves the overall problem better, and I can work on the namespacing issue we discussed above. 

The change does look like it's failing the tests though and I'm not sure why. It looks like there may be an issue with the `backreferences = true` (?)

Reply:
I do also have a PR up on the documentation for the API Generator (there wasn't anything on the generators docs) and made an update to my previous API example that uses the generated API controller code, instead of what I had handwritten (before I knew there was an API generator) https://github.com/amberframework/docs/pull/167/files